packages/synapse-react-client/src/components/SynapseChat/SynapseChat.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { KeyboardEventHandler, useEffect, useState } from 'react'
+import {
+  Box,
+  List,
+  Paper,
+  Typography,
+  FormControl,
+  InputAdornment,
+  IconButton,
+} from '@mui/material'
+import { useTheme } from '@mui/material'
+import { ColorPartial } from '@mui/material/styles/createPalette'
+import { OutlinedInput } from '@mui/material'
+import { ArrowUpward } from '@mui/icons-material'
+import SynapseChatInteraction from './SynapseChatInteraction'
+import { SkeletonParagraph } from '../Skeleton'
+import { useCreateAgentSession } from 'src/synapse-queries/chat/useChat'
+import { AgentAccessLevel } from '@sage-bionetworks/synapse-types'
+
+export type SynapseChatProps = {
+  initialMessage: string
+  agentId?: string // if provided, use this agent
+}
+
+export const SynapseChat: React.FunctionComponent<SynapseChatProps> = ({
+  initialMessage,
+  agentId,
+}) => {
+  // Create a new Agent Session on mount, add ability to initialize this component with a starting message from the user.
+  const [agentSessionId, setAgentSessionId] = useState('')
+  const { mutate: createAgentSession } = useCreateAgentSession({
+    onSuccess: newAgentSession => {
+      setAgentSessionId(newAgentSession.sessionId)
+    },
+  })
+  const theme = useTheme()
+  const [messages, setMessages] = useState([initialMessage])
```

Comment:
I am very surprised that the set of messages is not stored and provided by the server

Reply:
Good point.  Added this comment to the LLM doc (asking John):
> Nick brought up a good point after looking at my client prototype implementation.  Will there be a service call to ask for all existing request chatText and response responseText values in the history of this chat?  Right now I’m storing these on the client as the conversation happens, but if the client is eventually supposed to restore a previous session, I think this history is needed.